tests/base/src/test/java/org/wildfly/security/ssl/SSLv2HelloAuthenticationTest.java

```diff
@@ -0,0 +1,348 @@
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocket;
+
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedKeyManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.AuthenticationContextConfigurationClient;
+import org.wildfly.security.auth.realm.KeyStoreBackedSecurityRealm;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.SecurityIdentity;
+import org.wildfly.security.auth.server.SecurityRealm;
+import org.wildfly.security.permission.PermissionVerifier;
+import org.wildfly.security.x500.principal.X500AttributePrincipalDecoder;
+import org.wildfly.security.x500.cert.BasicConstraintsExtension;
+import org.wildfly.security.x500.cert.SelfSignedX509CertificateAndSigningKey;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+/**
+ * Simple test case to test two-way SSL using SSLv2Hello.
+ *
+ * @author <a href="mailto:szaldana@redhat">Sonia Zaldana</a>
+ */
+public class SSLv2HelloAuthenticationTest {
+
+    private static final boolean IS_IBM = System.getProperty("java.vendor").contains("IBM");
+    private static final char[] PASSWORD = "Elytron".toCharArray();
+    private static final String CA_JKS_LOCATION = "./target/test-classes/ca/jks";
+    private static File ladybirdFile = null;
+    private static File scarabFile = null;
+    private static File beetlesFile = null;
+    private static File trustFile = null;
+    private static File workingDirCA = null;
+
+    @BeforeClass
+    public static void setUp() throws Exception{
+        workingDirCA = new File(CA_JKS_LOCATION);
+        if (!workingDirCA.exists()) {
+            workingDirCA.mkdirs();
+        }
+
+        ladybirdFile = new File(workingDirCA,"ladybird.keystore");
+        scarabFile = new File(workingDirCA,"scarab.keystore");
+        beetlesFile = new File(workingDirCA,"beetles.keystore");
+        trustFile = new File(workingDirCA,"ca.truststore");
+
+        createKeyStores(ladybirdFile, scarabFile, beetlesFile, trustFile);
+    }
+
+
+    @AfterClass
+    public static void cleanUp(){
+        ladybirdFile.delete();
+        ladybirdFile = null;
+        scarabFile.delete();
+        scarabFile = null;
+        beetlesFile.delete();
+        beetlesFile = null;
+        trustFile.delete();
+        trustFile = null;
+        workingDirCA.delete();
+        workingDirCA = null;
+    }
+
+    @Test
+    public void testTwoWaySSLv2Hello() throws Exception {
+
```

Comment:
Looks like an assumption similar to the one in the WildFly Core test cases should be added here related to IBM JDK since this won't run successfully there, right?

Reply:
fixed